packages/context-menu/src/vaadin-menu-overlay-mixin.js

```diff
@@ -116,4 +120,48 @@ export const MenuOverlayMixin = (superClass) =>
         }
       }
     }
+
+    /**
+     * Override method inherited from `OverlayFocusMixin` to disable
+     * focus restoration on sub-menu overlay close. Focus should
+     * be only restored when the root menu closes.
+     *
+     * @protected
+     * @override
+     * @return {boolean}
+     */
+    _shouldRestoreFocus() {
+      if (this.parentOverlay) {
+        // Do not restore focus on sub-menu close.
+        return false;
+      }
+
+      return super._shouldRestoreFocus();
+    }
```

Comment:
The current set of tests doesn't check on this behavior, so if I remove this method override, the tests don't fail. I propose adding a test with a menu with <kbd>Enter</kbd> being triggered from a submenu item.

Reply:
Good catch. I added tests and, along the way, fixed another issue where it failed to restore focus on outside click.